### ISSUE TITLE
optimizer.step() does not check if the parameter has gradient

### ISSUE THREAD

**Issue by linyuhongg** [zou-group/textgrad#100]
Currently, the optimizer does not check whether the parameters have gradients when calling `.step()`, It could be valuable to return a warning.

**Comment by mertyg** [comment]
great point.. we should add this in the next release. thank you @linyuhongg !
